sagemaker-spark/pyspark_mnist/pyspark_mnist_kmeans.ipynb

```diff
@@ -0,0 +1,285 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.\n",
+    "\n",
+    "Licensed under the Apache License, Version 2.0 (the \"License\").\n",
+    "You may not use this file except in compliance with the License.\n",
+    "A copy of the License is located at\n",
+    " \n",
+    "  http://aws.amazon.com/apache2.0/\n",
+    "\n",
+    "or in the \"license\" file accompanying this file. This file is distributed\n",
+    "on an \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either\n",
+    "express or implied. See the License for the specific language governing\n",
+    "permissions and limitations under the License."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# SageMaker PySpark K-Means Clustering MNIST Example\n",
+    "\n",
+    "1. [Introduction](#Introduction)\n",
+    "2. [Loading the Data](#Loading-the-Data)\n",
+    "3. [Training and Hosting a Model](#Training-and-Hosting-a-Model)\n",
+    "4. [Inference](#Inference)\n",
+    "5. [More on SageMaker Spark](#More-on-SageMaker-Spark)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "This notebook will show how to classify handwritten digits using the K-Means clustering algorithm through the SageMaker PySpark library. We will train on Amazon SageMaker using K-Means clustering on the MNIST dataset, host the trained model on Amazon SageMaker, and then make predictions against that hosted model.\n",
+    "\n",
+    "You can visit SageMaker Spark's Github repository at https://github.com/aws/sagemaker-spark for more about SageMaker Spark.\n",
+    "\n",
+    "We will train on Amazon SageMaker using the KMeans Clustering on the MNIST dataset, host the trained model on Amazon SageMaker, and then make predictions against that hosted model.\n",
+    "\n",
+    "First, we load the MNIST dataset into a Spark Dataframe, which dataset is available in LibSVM format at\n",
```

Comment:
Can we add a sentence that provides clarity on where this Spark Dataframe is living? (i.e. It's in a Spark context on the Notebook Instance).  And maybe a sentence that mentions how this could be extended to connect with a Spark EMR cluster to orchestrate large data manipulations there.  Maybe include a link to this [post](https://aws.amazon.com/blogs/machine-learning/build-amazon-sagemaker-notebooks-backed-by-spark-in-amazon-emr/) too?

Reply:
Done and done